src/drivers/distance_sensor/lightware_sf45_serial/lightware_sf45_serial.cpp

```diff
@@ -762,10 +674,63 @@ void SF45LaserSerial::sf45_process_replies(float *distance_m)
 		break;
 	}
 }
+void SF45LaserSerial::_publish_obstacle_msg(hrt_abstime now)
+{
+	// This whole logic is here because the sensor has a faster loop time than collision prevention, and if we dont work with a timeout on sensor data
+	// we run into the problem of obstacle_distance messages being missed in collision prevention.
+	for (int i = 0; i < BIN_COUNT; i++) {
+		if (now - _data_timestamps[i] > SF45_MEAS_TIMEOUT) {
+			_obstacle_map_msg.distances[i] = UINT16_MAX;
+		}
+	}
 
-uint8_t SF45LaserSerial::sf45_convert_angle(const int16_t yaw)
+	_obstacle_map_msg.timestamp = now;
+	_obstacle_distance_pub.publish(_obstacle_map_msg);
+}
+
+void SF45LaserSerial::_handle_missed_bins(uint8_t current_bin, uint8_t previous_bin, uint16_t measurement,
+		hrt_abstime now)
 {
+	// if the sensor has its cycle delay configured for a low value like 5, it can happen that not every bin gets a measurement.
```

Comment:
```suggestion
	// if the sensor has its cycle delay configured for a low value like 5s, it can happen that not every bin gets a measurement.
```

Reply:
I guess..

Reply:
I haven't found a physical representation of that number, 
the docs state "The scan speed controls the delay between each scan position"
but with a scan speed of 5, we have 0.7 seconds between two measurements of the same angle. 
And "scan speed" is the name they give it in the i2c command list, "cycle delay" is what's used in their LightWare Studio software, as I assume people use the software to program it, I used this term.